### ISSUE TITLE
Quotation for values containing whitespace (eg. Unix shell variable assignments)

### ISSUE THREAD

**Issue by rwp0** [TLINDEN/Config-General#1]
Thank you for this module, it's useful as a general-purpose config parser. I especially like the `::Extended` interface.

One thing I noticed, it doesn't preserve the quotes around values even while all the values are having quotes.

For example, parsing FreeBSD [rc.conf](https://github.com/freebsd/freebsd-src/blob/main/libexec/rc/rc.conf) file which is basically Bourne shell value-to-variable assignments:

```sh
apache24_flags="-f /usr/local/apache/conf/httpd.conf"
static_arp_pairs="gwint gwpub"
```

The module removes the double quotation marks leaving this snippet as:

```sh
apache24_flags=-f /usr/local/apache/conf/httpd.conf
static_arp_pairs=gwint gwpub
```

Which makes the Unix shell fail to read them.

It could be useful if we had an option switch passed to the `new` method in order to quote all values, let's say as `-QuoteValues=1`.

We can also have `-QuoteType` as double (`""`) or single (`''`), but that's extra.

Thanks

**Comment by TLINDEN** [comment]
Hi,

this is indeed a great idea. I'll look into it as soon I find the time.

**Comment by rwp0** [comment]
> Hi,
> 
> this is indeed a great idea. I'll look into it as soon I find the time.

Thanks.

**Comment by rwp0** [comment]
Has there been any progress on this since?

**Comment by TLINDEN** [comment]
Sorry, not yet...

**Comment by TLINDEN** [comment]
Nope, while this fixes the issue, it breaks other stuff. Reopened, I'll add a flag.

**Comment by TLINDEN** [comment]
Now there's a new flag `-AlwaysQuoteOutput` which can be enabled to get the behavior, that values containing whitespace are being written with quotes around it.
